cmd/service_linux.go

```diff
@@ -0,0 +1,316 @@
+//go:build linux
+
+/*
+   Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+
+   Copyright © 2023-2024 Seagate Technology LLC and/or its Affiliates
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+*/
+
+package cmd
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+
+	"github.com/Seagate/cloudfuse/common"
+	"github.com/spf13/cobra"
+)
+
+type serviceOptions struct {
+	ConfigFile string
+	MountPath  string
+}
+
+var servOpts serviceOptions
+
+// Section defining all the command that we have in secure feature
+var serviceCmd = &cobra.Command{
+	Use:               "service",
+	Short:             "Manage cloudfuse startup process on Linux",
+	Long:              "Manage cloudfuse startup process on Linux",
+	SuggestFor:        []string{"ser", "serv"},
+	Example:           "cloudfuse service install",
+	FlagErrorHandling: cobra.ExitOnError,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return errors.New("missing command options\n\nDid you mean this?\n\tcloudfuse service install\n\nRun 'cloudfuse service --help' for usage")
+	},
+}
+
+var installCmd = &cobra.Command{
+	Use:               "install [mount path] [config path]",
```

Comment:
Could you change this to use flags like mount-path and config-path instead of relying on order?